src/test/shared/sam/debugger/samDebugConfigProvider.test.ts

```diff
@@ -103,6 +108,11 @@ describe('AwsSamDebugConfigurationProvider', async () => {
             assert.deepStrictEqual(await debugConfigProvider.provideDebugConfigurations(undefined), [])
             // Workspace with no templates:
             assert.deepStrictEqual(await debugConfigProvider.provideDebugConfigurations(fakeWorkspaceFolder), [])
+
+            // Malformed template.yaml:
+            await strToYamlFile('bogus', tempFile.fsPath)
```

Comment:
JS-YAML still parses this as valid YAML. Use https://github.com/aws/aws-toolkit-vscode/blob/ab650f54d0a4526aadcbff2979136f27ba735717/src/test/shared/cloudformation/cloudformationTestUtils.ts#L81 instead (this definitely doesn't parse)

Reply:
Thanks. We can have more insightful tests by checking the error popups or log messages explicitly. Will add a tracking ticket for that.

Reply:
I made that change, but then reverted it: I am not looking for unparsable YAML, but just a bad template.yaml. I.e. I want a semantic error, not a syntax error (`YAMLException`).